### PR TITLE
Service pipelines using variable groups from bedrock.yaml

### LINKED ISSUE #887: [spk] Service pipelines should only use project Variable group
**As a:** SPK user and developer

**I want:** service pipelines to only utilize variable groups for variables
instead of those baked into the yaml

**So that:** the utilization of variable groups across lifecycle and service
pipelines is standardized across the project

**Describe the solution you'd like:**

- [x] Refactor `spk service create` to output azure-piplines.yaml that consume
      the variable groups created in `spk project cvg`
- [x] Refactor `spk service create` to remove the `--variable-group-name <variable-group-name` flag, as spk should only use variable groups defined under the project's `bedrock.yaml
- [x] Check if user executed `spk project init` and `spk project cvg` commands before `spk service create` command for dependencies

**Acceptance Criteria:**

refer to task list above

**Describe alternatives you've considered:**

**Additional context:**

[Project commands are now expected to run in order](https://github.com/microsoft/bedrock/issues/886)

**Does this require updates to documentation?:** yes

### PR DESCRIPTION
**Pull request by sarath-p**
- Closes https://github.com/microsoft/bedrock/issues/887
- Checks if user ran `spk project init` and `spk project cvg` commands before this `spk service create`

### CODE REVIEW

**Comment by dennisseah** [comment]
We shall have conflict with other PR that I have https://github.com/CatalystCode/spk/pull/215
I have moved isBedrockFileExist to a new helper file where it contains all the bedrock related functions.

**Comment by sarath-p** [comment]
> We shall have conflict with other PR that I have #215
> I have moved isBedrockFileExist to a new helper file where it contains all the bedrock related functions.

The function code also changed. We can merge through master branch.

File: src/config.ts [hunk 1]
```diff
@@ -294,3 +294,31 @@ export const saveConfiguration = async (
...
+  try {
+    bedrockFile = await BedrockAsync(absProjectPath);
+  } catch (error) {
+    logger.error(error);
+    return [false, undefined];
+  }
+
+  logger.debug(`variableGroups length: ${bedrockFile?.variableGroups?.length}`);
+  logger.verbose(`bedrockFile: \n ${bedrockFile}`);
+  return [true, bedrockFile?.variableGroups?.length];
```
Comment (dennisseah) [inline src/config.ts]: you are overloading this function. I hope that you can return `true` or `false` here. and then do variableGroups length check in the caller function. When we overload function, you can see that you needed to return things like undefined. thanks.

Comment (sarath-p) [reply src/config.ts]: It is a deep check i.e. not only a file exist, but need to make sure variable groups exist in the file. In this scenario, the caller does't need the file contents and true/false is sufficient. Perhaps we can change the second one to `boolean` and remove the `undefined`.  Let me know your thoughts on this.

Comment (dennisseah) [reply src/config.ts]: I am misled by the function name, isBedrockFileValid. As the function name suggest, I thought that we are checking if bedrock file exist or not :-) I think we can do this.
```
interface IBedrockFileStat {
  exist: boolean;
  hasVariableGroups: boolean;
}

export const bedrockFileStat = async (
  rootProjectPath: string
): Promise<IBedrockFileStat> => {
  if (!rootProjectPath) {
    throw new Error("Project root path is not valid");
  }

  const absProjectPath = path.resolve(rootProjectPath);

  let bedrockFile: IBedrockFile | undefined;

  try {
    const bedrockFile = await BedrockAsync(absProjectPath);
    return {
      exist: true,
      hasVariableGroups: bedrockFile?.variableGroups?.length
    }
  } catch (error) {
    logger.error(error);
    return {
      exist: false,
    }
  }
}
```

this enables other team members to reuse and enhance this function get more stats from bedrock file.
and we do not return array, we return a well defined interface.

what do you think?

Comment (sarath-p) [reply src/config.ts]: Yes, the name was misleading. I thought about it and am going to refactor it to check whether dependency commands were executed.  One of the goal it to centralize the error messages too so they won't be repeated.

Comment (sarath-p) [reply src/config.ts]: Discussed with @mtarng and the decided to keep it as `bedrock file info` and use it to validate the project commands order.

Comment (dennisseah) [reply src/config.ts]: sorry, I do not understand `keep it as bedrock file info`. Do you mind elaborate?

Comment (sarath-p) [reply src/config.ts]: I was thinking to implement command sequence validator using this logic, but decided to keep the method name as bedrock file that may be used for other checks in the future.

Comment (dennisseah) [reply src/config.ts]: ok

**Review by dennisseah — APPROVED** [review]

File: src/commands/project/create-variable-group.ts [hunk 2]
```diff
@@ -76,72 +82,80 @@ export const execute = async (
...
-        logger.info(
-          "Successfully created a variable group in Azure DevOps project!"
-        );
-        await exitFn(0);
-      }
-    } catch (err) {
-      logger.error(`Error occurred while creating variable group`);
-      logger.error(err);
-      await exitFn(1);
+    const bedrockFileIno = await bedrockFileInfo(projectPath);
```
Comment (mtarng) [inline src/commands/project/create-variable-group.ts]: Typo?
>bedrockFileIno

or bedrockFileInfo

Comment (sarath-p) [reply src/commands/project/create-variable-group.ts]: yes, let me fix it

**Review by mtarng — APPROVED** [review]

**Pull request merged**